app/controllers/users/confirmations_controller.rb

```diff
@@ -112,7 +112,12 @@ def after_confirmation_path_for(resource)
     end
 
     def process_confirmed_user
-      analytics.track_event('Email changed and confirmed', @confirmable)
+      if @confirmable.previous_changes.key?('email')
```

Comment:
This check is not necessary. If this method is called, it means the user just confirmed their new email.